### ISSUE TITLE
Augmentations and # of training epochs

### ISSUE THREAD

**Issue by georgestein** [emadeldeen24/TS-TCC#20]
While recently implementing TS-TCC I ran into the same issue as one raised here by vamsi231297 a few years ago that self-supervised TS-TCC only augments once at the beginning of training. 

To my knowledge all other methods randomly augment each data sample at each epoch, and not just once at the start. See for example Mocov2 (https://github.com/facebookresearch/moco/blob/main/main_moco.py#L351). This is the usual way to prevent overfitting & increase generalization.

I implemented augmenting each epoch rather than just once at the start, and found that it increased the performance of TS-TCC on the HAR dataset. Linear probe after self supervised phase went from 0.9160 accuracy to 0.9294. 9.294 is higher than the values and errorbars reported in Table 2 of the paper.

Saying that, I am also finding that training for more epochs with the rest of the parameters unchanged results in better performance than shown in Table 2 of the paper. E.g. 1000 epochs of supervised training gets accuracy=0.9408. Changing the learning rate also can get better accuracies. 

So I would not take my test at face value as an improvement of TS-TCC over supervised as it seems likely that my results and those of Table 2 are highly dependent on hyperparameters and seeds.

**Comment by emadeldeen24** [comment]
Hi @georgestein, thanks for your comment.

I agree with your comment about generating augmented views on the fly during training. I resolved this concern in my other work [eval_ssl_ssc](https://github.com/emadeldeen24/eval_ssl_ssc), which also has TS-TCC implementation. You are welcome to pull your updates to this repo to merge it.

Also, I agree that changing the training hyperparameters can affect the results. Therefore, I recommend that if you want to compare some methods, you should train them with the same settings to ensure a fair comparison.

Thanks again for experimenting and reporting.
